tests/run-pass/close-stderr.rs

```diff
@@ -0,0 +1,16 @@
+// compile-flags: -Zmiri-disable-isolation
+
+#![feature(rustc_private)]
+
+extern crate libc;
+
+fn main() -> std::io::Result<()> {
+    let bytes = b"hello\n";
+    unsafe {
+        libc::write(2, bytes.as_ptr() as *const libc::c_void, 6);
+        libc::close(2);
+        libc::write(2, bytes.as_ptr() as *const libc::c_void, 6);
```

Comment:
Shouldn't this error? Please check the error code.

Also please merge all these new run-pass tests into a single file, `fs_libc.rs`, with one function per test.

Reply:
Added an assertion for this. And merged this and other tests into a single file.